From 16548f6dd25c4fbafba40a0a0613dd500fbf1fbb Mon Sep 17 00:00:00 2001
From: Jan Lieskovsky <jlieskov@redhat.com>
Date: Tue, 15 Dec 2015 12:02:21 +0100
Subject: [PATCH] [BugFix] [RHEL/6] Update the provided kickstart files for
 RHEL/6 product to utilize services of the internal "scap-security-guide" RPM
 package (rather than git clone upstream repository content)

---
 RHEL/6/kickstart/ssg-rhel6-pci-dss-with-gui-ks.cfg | 42 +++++---------------
 RHEL/6/kickstart/ssg-rhel6-stig-ks.cfg             | 43 ++++++---------------
 .../ssg-rhel6-usgcb-server-with-gui-ks.cfg         | 45 ++++++----------------
 3 files changed, 33 insertions(+), 97 deletions(-)

diff --git a/RHEL/6/kickstart/ssg-rhel6-pci-dss-with-gui-ks.cfg b/RHEL/6/kickstart/ssg-rhel6-pci-dss-with-gui-ks.cfg
index 215de4c..9f54758 100644
--- a/RHEL/6/kickstart/ssg-rhel6-pci-dss-with-gui-ks.cfg
+++ b/RHEL/6/kickstart/ssg-rhel6-pci-dss-with-gui-ks.cfg
@@ -1,6 +1,6 @@
 # SCAP Security Guide PCI-DSS profile kickstart for Red Hat Enterprise Linux 6 Server
-# Version: 0.0.1
-# Date: 2015-04-11
+# Version: 0.0.2
+# Date: 2015-12-15
 #
 # Based on:
 # http://fedoraproject.org/wiki/Anaconda/Kickstart
@@ -148,53 +148,31 @@ logvol swap --name=lv_swap --vgname=vg_sys --size=2016
 @web-server
 @x11
 
-# Install git package so we can clone SCAP Security Guide repository below
-git
-
 # Install selected additional packages (required by PCI DSS profile for Red Hat
 # Enterprise Linux 6 Server)
 # CCE-27024-9: Install AIDE
 aide
 
-# Install openscap-utils so it's possible to perform remediation once the
+# Install openscap-scanner so it's possible to perform remediation once the
 # installation is complete
 openscap-scanner
 
+# Install the scap-security-guide package
+scap-security-guide
+
 %end # End of %packages section
 
 %post --log /root/oscap.log
 
-SSG_GIT_CONTENT="/root/scap-security-guide"
-OSCAP_GIT_CONTENT="/root/oscap"
-SSG_GIT_RHEL6="$SSG_GIT_CONTENT/RHEL/6"
-SSG_GIT_URI="https://github.com/OpenSCAP/scap-security-guide.git"
-SSG_SHARE_DIR="/usr/share/scap-security-guide"
-
-# Retrieve the most recent SCAP Security Guide repository content & build
-# the RHEL-6 benchmark.
-#
-# NOTE: Pointing to upstream SCAP Security Guide repository. Downstream/stable editions
-#       (e.g. those to ship natively with Red Hat Enterprise Linux 6) will be updated to
-#       point to natively included scap-security-guide RPM package.
-
-git clone $SSG_GIT_URI $SSG_GIT_CONTENT
-cd $SSG_GIT_RHEL6
-make
-
-# Copy the remediation functions library to the system too
-mkdir -p $SSG_SHARE_DIR
-cp -a $SSG_GIT_CONTENT/shared/remediations/bash/templates/remediation_functions $SSG_SHARE_DIR/remediation_functions
-
 # Perform post installation system remediation according to the PCI DSS profile
 # for Red Hat Enterprise Linux 6 Server via the oscap tool To create a system
 # compliant against different RHEL-6 SCAP Security Guide profile specify
 # selected profile name after the --profile oscap tool option
 
-oscap xccdf eval --remediate --profile pci-dss --report /root/oscap_pci-dss_remediation_report.html \
-"$SSG_GIT_RHEL6/output/ssg-rhel6-xccdf.xml"
-
-# Delete the retrieved GIT content
-rm -rf $SSG_GIT_CONTENT
+oscap xccdf eval --remediate --profile pci-dss \
+--results /root/oscap_pci_dss_remediation_result.xml \
+--report /root/oscap_pci_dss_remediation_report.html \
+/usr/share/xml/scap/ssg/content/ssg-rhel6-xccdf.xml
 
 %end # End of %post section
 
diff --git a/RHEL/6/kickstart/ssg-rhel6-stig-ks.cfg b/RHEL/6/kickstart/ssg-rhel6-stig-ks.cfg
index ecdf414..957ba2b 100644
--- a/RHEL/6/kickstart/ssg-rhel6-stig-ks.cfg
+++ b/RHEL/6/kickstart/ssg-rhel6-stig-ks.cfg
@@ -1,6 +1,6 @@
 # SCAP Security Guide DoD STIG profile kickstart for Red Hat Enterprise Linux 6 Server
-# Version: 0.0.1
-# Date: 2015-04-08
+# Version: 0.0.2
+# Date: 2015-12-15
 #
 # Based on:
 # http://fedoraproject.org/wiki/Anaconda/Kickstart
@@ -348,51 +348,30 @@ usbutils
 libcap-devel.i686
 libselinux-devel.i686
 
-# Install git package so we can clone SCAP Security Guide repository below
-git
-
 # Install selected additional packages (required by DoD STIG profile for Red Hat Enterprise Linux 6 Server)
 # CCE-27024-9: Install AIDE
 aide
 
-# Install openscap-utils so it's possible to perform remediation once the
+# Install openscap-scanner so it's possible to perform remediation once the
 # installation is complete
-openscap-utils
+openscap-scanner
+
+# Install the scap-security-guide package
+scap-security-guide
 
 %end # End of %packages section
 
 %post --log /root/oscap.log
 
-SSG_GIT_CONTENT="/root/scap-security-guide"
-SSG_GIT_RHEL6="$SSG_GIT_CONTENT/RHEL/6"
-SSG_GIT_URI="https://github.com/OpenSCAP/scap-security-guide.git"
-SSG_SHARE_DIR="/usr/share/scap-security-guide"
-
-# Retrieve the most recent SCAP Security Guide repository content & build
-# the RHEL-6 benchmark.
-
-# NOTE: Pointing to upstream SCAP Security Guide repository. Downstream/stable editions
-#       (e.g. those to ship natively with Red Hat Enterprise Linux 6) will be updated to
-#       point to natively included scap-security-guide RPM package.
-
-git clone $SSG_GIT_URI $SSG_GIT_CONTENT
-cd $SSG_GIT_RHEL6
-make
-
-# Copy the remediation functions library to the system too
-mkdir -p $SSG_SHARE_DIR
-cp -a $SSG_GIT_CONTENT/shared/remediations/bash/templates/remediation_functions $SSG_SHARE_DIR/remediation_functions
-
 # Perform post installation system remediation according to the DoD STIG profile
 # for Red Hat Enterprise Linux 6 Server via the oscap tool
 # To create a system compliant against different RHEL-6 SCAP Security Guide profile specify selected
 # profile name after the --profile oscap tool option
 
-oscap xccdf eval --remediate --profile stig-rhel6-server-upstream --report /root/oscap_stig_remediation_report.html \
-"$SSG_GIT_RHEL6/output/ssg-rhel6-xccdf.xml"
-
-# Delete the retrieved GIT content
-rm -rf $SSG_GIT_CONTENT
+oscap xccdf eval --remediate --profile stig-rhel6-server-upstream \
+--results /root/oscap_stig_remediation_result.xml \
+--report /root/oscap_stig_remediation_report.html \
+/usr/share/xml/scap/ssg/content/ssg-rhel6-xccdf.xml
 
 %end # End of %post section
 
diff --git a/RHEL/6/kickstart/ssg-rhel6-usgcb-server-with-gui-ks.cfg b/RHEL/6/kickstart/ssg-rhel6-usgcb-server-with-gui-ks.cfg
index 213c56f..4894431 100644
--- a/RHEL/6/kickstart/ssg-rhel6-usgcb-server-with-gui-ks.cfg
+++ b/RHEL/6/kickstart/ssg-rhel6-usgcb-server-with-gui-ks.cfg
@@ -1,6 +1,6 @@
 # SCAP Security Guide USGCB profile kickstart for Red Hat Enterprise Linux 6 Server
-# Version: 0.0.1
-# Date: 2014-10-23
+# Version: 0.0.2
+# Date: 2015-12-15
 #
 # Based on:
 # http://fedoraproject.org/wiki/Anaconda/Kickstart
@@ -17,7 +17,7 @@ install
 # Install from an installation tree on a remote server via FTP or HTTP:
 # --url		the URL to install from
 #
-url --url=http://192.168.122.1/image
+#url --url=http://192.168.122.1/image
 #
 # Other possible / supported installation methods:
 # * install from the first CD-ROM/DVD drive on the system:
@@ -143,50 +143,29 @@ logvol swap --name=lv_swap --vgname=VolGroup --size=2016
 @web-server
 @x11
 
-# Install git package so we can clone SCAP Security Guide repository below
-git
-
 # Install selected additional packages (required by USGCB profile)
 # CCE-27024-9: Install AIDE
 aide
 
-# Install openscap-utils so it's possible to perform remediation once the
+# Install openscap-scanner so it's possible to perform remediation once the
 # installation is complete
-openscap-utils
+openscap-scanner
+
+# Install the scap-security-guide package
+scap-security-guide
 
 %end # End of %packages section
 
 %post --log /root/oscap.log
 
-SSG_GIT_CONTENT="/root/scap-security-guide"
-SSG_GIT_RHEL6="$SSG_GIT_CONTENT/RHEL/6"
-SSG_GIT_URI="https://github.com/OpenSCAP/scap-security-guide.git"
-SSG_SHARE_DIR="/usr/share/scap-security-guide"
-
-# Retrieve the most recent SCAP Security Guide repository content & build
-# the RHEL-6 benchmark.
-
-# NOTE: Pointing to upstream SCAP Security Guide repository. Downstream/stable editions
-#       (e.g. those to ship natively with Red Hat Enterprise Linux 6) will be updated to
-#       point to natively included scap-security-guide RPM package.
-
-git clone $SSG_GIT_URI $SSG_GIT_CONTENT
-cd $SSG_GIT_RHEL6
-make
-
-# Copy the remediation functions library to the system too
-mkdir -p $SSG_SHARE_DIR
-cp -a $SSG_GIT_CONTENT/shared/remediations/bash/templates/remediation_functions $SSG_SHARE_DIR/remediation_functions
-
 # Perform post installation system remediation according to the USGCB profile via the oscap tool
 # To create a system compliant against different RHEL-6 SCAP Security Guide profile specify selected
 # profile name after the --profile oscap tool option
 
-oscap xccdf eval --remediate --profile usgcb-rhel6-server --report /root/oscap_usgcb_remediation_report.html \
-"$SSG_GIT_RHEL6/output/ssg-rhel6-xccdf.xml"
-
-# Delete the retrieved GIT content
-rm -rf $SSG_GIT_CONTENT
+oscap xccdf eval --remediate --profile usgcb-rhel6-server \
+--results /root/oscap_usgcb_remediation_result.xml \
+--report /root/oscap_usgcb_remediation_report.html \
+/usr/share/xml/scap/ssg/content/ssg-rhel6-xccdf.xml
 
 %end # End of %post section
 
-- 
2.4.3

